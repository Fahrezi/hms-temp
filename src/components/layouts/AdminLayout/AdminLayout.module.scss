.layout {
  & {
    min-height: 100vh;
    display: flex;
    flex-direction: column;
  }

  &__header {
    height: 4rem;
    border-bottom: 1px solid #e5e7eb;
    display: flex;
    align-items: center;
    padding: 0 2rem;
    background: white;

    &__button {
      margin-left: auto;
    }
  }

  &__main {
    flex: 1;
    background-color: #f3f4f6;
  }

  &__footer {
    height: 4rem;
    border-top: 1px solid #e5e7eb;
    display: flex;
    align-items: center;
    padding: 0 2rem;
    background: white;
  }
}
