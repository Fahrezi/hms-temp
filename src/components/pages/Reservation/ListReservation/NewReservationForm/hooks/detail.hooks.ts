import { useCallback, useEffect, useState } from "react"

type List = {
  label: string,
  value: string
};

type ListResponse = {
  name: string,
  id: string
};

const BASE_URL = "http://localhost:5174";

export const useDetail = () => {
  const [countryList, setCountryList] = useState<List[]>([]);
  const [citiesList, setCitiesList] = useState<List[]>([]);
  const [nationalityList, setNationalityList] = useState<List[]>([]);
  const [languageList, setLanguageList] = useState<List[]>([]);

  const getData = useCallback(async () => {
    const countryResult = await fetch(`${BASE_URL}/countries  `).then(res => res.json());
    const citiesResult = await fetch(`${BASE_URL}/cities`).then(res => res.json());
    const nationalityResult = await fetch(`${BASE_URL}/nationality`).then(res => res.json());
    const languageResult = await fetch(`${BASE_URL}/language`).then(res => res.json());

    setCountryList(countryResult.map((country: ListResponse) => ({ label: country.name, value: country.id })));
    setCitiesList(citiesResult.map((city: ListResponse) => ({ label: city.name, value: city.id })));
    setNationalityList(nationalityResult.map((nation: ListResponse) => ({ label: nation.name, value: nation.id })));
    setLanguageList(languageResult.map((lang: ListResponse) => ({ label: lang.name, value: lang.id })));
  }, []);

  useEffect(() => {
    getData();
  }, [getData]);


  return {
    countryList,
    citiesList,
    nationalityList,
    languageList
  }
}