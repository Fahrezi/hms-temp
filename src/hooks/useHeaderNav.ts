import { useContext } from 'react';

import HeaderNavContext from '@/contexts/HeaderNavContext';


export const useHeaderNav = () => {
  const context = useContext(HeaderNavContext);
  if (context === undefined) {
    throw new Error('useHeaderNav must be used within an HeaderNav');
  }
  return context;
};
