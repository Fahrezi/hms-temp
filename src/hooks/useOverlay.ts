import { useContext } from 'react';

import OverlayContext from '@/contexts/OverlayContext';

export const useOverlay = () => {
  const context = useContext(OverlayContext);
  if (context === undefined) {
    throw new Error('useOverlay must be used within an HeaderNav');
  }
  return context;
};
