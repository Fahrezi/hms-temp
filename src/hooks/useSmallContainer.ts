import { useEffect } from 'react';

import getEnvValue from '@/utils/env';
import { ENVIRONMENT } from '@/constants/envs';

const SMALL_CONTAINER_CLASS = 'small-container';

const useSmallContainer = () => {
  useEffect(() => {
    const isSmallContainer = getEnvValue(ENVIRONMENT.ENABLE_SMALL_CONTAINER) === 'true';

    if (isSmallContainer) {
      document.body.classList.add(SMALL_CONTAINER_CLASS);
    } else {
      document.body.classList.remove(SMALL_CONTAINER_CLASS);
    }

    return () => {
      document.body.classList.remove(SMALL_CONTAINER_CLASS);
    };
  }, []);
};

export default useSmallContainer;
