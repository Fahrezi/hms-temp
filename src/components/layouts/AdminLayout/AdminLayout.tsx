import { Outlet } from 'react-router-dom';

import { useAuth } from '@/hooks/useAuth';

import styles from './AdminLayout.module.scss';

const AdminLayout = () => {
  const { logout } = useAuth();

  return (
    <div className={styles.layout}>
      <header className={styles.layout__header}>
        <button onClick={logout} className={styles.layout__header__button}>
          Log Out
        </button>
      </header>

      <main className={styles.layout__main}>
        <Outlet />
      </main>

      <footer className={styles.layout__footer}>{/* Add your footer content */}</footer>
    </div>
  );
};

export default AdminLayout;
