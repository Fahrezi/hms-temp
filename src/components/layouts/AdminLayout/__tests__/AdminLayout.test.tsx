import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { vi } from 'vitest';

import AdminLayout from '@/components/layouts/AdminLayout';

import { useAuth } from '@/hooks/useAuth';

vi.mock('@/hooks/useAuth', () => ({
  useAuth: vi.fn(),
}));

const mockUseAuth = useAuth as unknown as ReturnType<typeof vi.fn>;

describe('AdminLayout Component', () => {
  it('renders the layout correctly', () => {
    const mockLogout = vi.fn();
    mockUseAuth.mockReturnValue({ logout: mockLogout });

    const { container } = render(
      <MemoryRouter>
        <AdminLayout />
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });

  it('calls logout function when logout button is clicked', () => {
    const mockLogout = vi.fn();
    mockUseAuth.mockReturnValue({ logout: mockLogout });

    render(
      <MemoryRouter>
        <AdminLayout />
      </MemoryRouter>,
    );

    const logoutButton = screen.getByRole('button', { name: /log out/i });
    fireEvent.click(logoutButton);

    expect(mockLogout).toHaveBeenCalledTimes(1);
  });
});
