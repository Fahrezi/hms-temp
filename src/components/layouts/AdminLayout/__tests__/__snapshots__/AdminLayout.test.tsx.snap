// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AdminLayout Component > renders the layout correctly 1`] = `
<div>
  <div
    class="_layout_ef7806"
  >
    <header
      class="_layout__header_ef7806"
    >
      <button
        class="_layout__header__button_ef7806"
      >
        Log Out
      </button>
    </header>
    <main
      class="_layout__main_ef7806"
    />
    <footer
      class="_layout__footer_ef7806"
    />
  </div>
</div>
`;
